cpp/src/cuimage.cpp

```diff
@@ -143,21 +144,25 @@ std::unique_ptr<plugin::ImageFormat> CuImage::image_format_plugins_ = std::make_
 CuImage::CuImage(const filesystem::Path& path)
 {
     PROF_SCOPED_RANGE(PROF_EVENT_P(cuimage_cuimage, 1));
-    //    printf("[cuCIM] CuImage::CuImage(filesystem::Path path)\n");
+
     ensure_init();
     image_format_ = image_format_plugins_->detect_image_format(path);
 
     // TODO: need to detect available format for the file path
     {
         PROF_SCOPED_RANGE(PROF_EVENT(cuimage_cuimage_open));
-        file_handle_ = image_format_->image_parser.open(path.c_str());
+        std::shared_ptr<CuCIMFileHandle>* file_handle_shared =
+            reinterpret_cast<std::shared_ptr<CuCIMFileHandle>*>(image_format_->image_parser.open(path.c_str()));
```

Comment:
Curious why this is `shared_ptr<...>*`?

Reply:
Please see the answer https://github.com/rapidsai/cucim/pull/191#discussion_r793135703